domain/credential/service/service.go

```diff
@@ -60,6 +60,10 @@ type State interface {
 	// for a given owner.
 	AllCloudCredentialsForOwner(ctx context.Context, owner user.Name) (map[corecredential.Key]credential.CloudCredentialResult, error)
 
+	// ValidCloudCredentialsForOwner returns the owner's cloud credentials for a given
+	// cloud, keyed by credential name.
+	ValidCloudCredentialsForOwner(ctx context.Context, owner user.Name, cloudName string) (map[string]credential.CloudCredentialResult, error)
```

Comment:
Valid should be implied in the contract and not the func name. Credentials should already be valid if they are in the database. I would suggest `DefaultUserCloudCredential`. You can then only supply a credential if it is the only credential available in state.

Reply:
renamed